### PR TITLE
AMD-902 [frontend] feat: pipeline configuration need to retain modified data when user change time range or reverified

### PR DESCRIPTION
**Pull request by doujiao-001**
## Summary
pipeline configuration need to retain modified data when user change time range or reverified
## Before

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## Note

_Null_

### CODE REVIEW

File: frontend/__tests__/containers/MetricsStep/Crews.test.tsx [hunk 1]
```diff
@@ -46,13 +46,18 @@ describe('Crew', () => {
     expect(screen.getByText(mockTitle)).toBeInTheDocument();
   });
 
-  it('should selected all options by default when initializing', () => {
+  it('should selected all options by default when initializing  as type is board', () => {
```
Comment (guzhongren) [inline frontend/__tests__/containers/MetricsStep/Crews.test.tsx:49]: as -> given

File: frontend/__tests__/containers/MetricsStep/Crews.test.tsx [hunk 2]
```diff
@@ -46,13 +46,18 @@ describe('Crew', () => {
...
     setup();
 
     expect(screen.getByRole('button', { name: 'crew A' })).toBeInTheDocument();
     expect(screen.getByRole('button', { name: 'crew B' })).toBeInTheDocument();
   });
+  it('should selected all options by default when initializing as type is other', () => {
```
Comment (guzhongren) [inline frontend/__tests__/containers/MetricsStep/Crews.test.tsx:55]: as 是什么？从哪来的？为什么？

File: frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/DeploymentFrequencySettings.test.tsx [hunk 3]
```diff
@@ -80,15 +86,45 @@ jest.mock('@src/hooks/useGetPipelineToolInfoEffect', () => ({
...
+    mockGetPipelineToolInfoSpy = mockGetPipelineToolInfoOkResponse;
+  });
+
+  it('should show crew settings when select pipelineName', async () => {
+    mockSelectPipelineNames = ['Heartbeat'];
+    const { getAllByRole, getByRole, debug } = await setup();
+    await act(async () => {
+      await userEvent.click(getAllByRole('button', { name: LIST_OPEN })[0]);
+    });
+    debug();
```
Comment (guzhongren) [inline frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/DeploymentFrequencySettings.test.tsx:113]: debug?

File: frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/DeploymentFrequencySettings.test.tsx [hunk 4]
```diff
@@ -122,14 +158,21 @@ describe('DeploymentFrequencySettings', () => {
...
     const listBox = within(getByRole('listbox'));
     await act(async () => {
-      await userEvent.click(listBox.getByText('mockOrgName'));
+      await userEvent.click(listBox.getByText('mockOrgName2'));
     });
 
     expect(updateDeploymentFrequencySettings).toHaveBeenCalledTimes(1);
   });
 
+  it('show render crews component when totalPipelineNumber is equal loadingCompletedNumber', () => {
```
Comment (guzhongren) [inline frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/DeploymentFrequencySettings.test.tsx:167]: when -> given
is equal ??? 

Please write test description with biz aspect, not tech aspect

File: frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/DeploymentFrequencySettings.test.tsx [hunk 5]
```diff
@@ -122,14 +158,21 @@ describe('DeploymentFrequencySettings', () => {
...
+    mockSelectShouldGetPipelineConfig = false;
+    setup();
+    expect(screen.getByText('Crew setting (optional)')).toBeInTheDocument();
+  });
```
Comment (guzhongren) [inline frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/DeploymentFrequencySettings.test.tsx:171]: add new line for `given when then`

File: frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/DeploymentFrequencySettings.test.tsx [hunk 6]
```diff
@@ -158,6 +202,15 @@ describe('DeploymentFrequencySettings', () => {
...
 
+  it('should not show show crews part given isFirstFetch is true', async () => {
+    mockGetPipelineToolInfoSpy = {
+      ...mockGetPipelineToolInfoOkResponse,
+      isFirstFetch: true,
+    };
+    setup();
+    expect(screen.queryByText('Crews Setting')).toBeNull();
+  });
+
```
Comment (guzhongren) [inline frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/DeploymentFrequencySettings.test.tsx:213]: add new line for given when then

**Review by guzhongren — CHANGES_REQUESTED** [review]
Q: 这个代码code diff 过么？

File: frontend/__tests__/containers/MetricsStep/Crews.test.tsx (hunk 1 above)
Comment (neomgb) [reply frontend/__tests__/containers/MetricsStep/Crews.test.tsx:49]: ok

File: frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/DeploymentFrequencySettings.test.tsx (hunk 3 above)
Comment (neomgb) [reply frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/DeploymentFrequencySettings.test.tsx:113]: removed

File: frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/DeploymentFrequencySettings.test.tsx (hunk 4 above)
Comment (neomgb) [reply frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/DeploymentFrequencySettings.test.tsx:167]: ok

File: frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/DeploymentFrequencySettings.test.tsx (hunk 5 above)
Comment (neomgb) [reply frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/DeploymentFrequencySettings.test.tsx:171]: ok

File: frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/DeploymentFrequencySettings.test.tsx (hunk 6 above)
Comment (neomgb) [reply frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/DeploymentFrequencySettings.test.tsx:213]: done

**Review by neomgb — COMMENTED** [review]
rename test  case descriptions

**Review by Leiqiuhong — APPROVED** [review]

**Review by neomgb — APPROVED** [review]

**Comment by neomgb** [comment]
> Q: 这个代码code diff 过么？
diff 的时候没太注意到，后续多注意一下 test case 的描述...

**Pull request closed without merging**